### PR TITLE
Fix the running average outdoor air temperature in the first day 

### PR DESCRIPTION
**Pull request by lgu1234**
Pull request overview
---------------------
This fix is to provide correct running average outdoor air temperature of the first day as Phase 1, addresses #5381  Here is what has been fixed in the CalcThermalComfortAdaptiveASH55 function of the ThermalComfort module.
1. Skip 9 lines when reading a weather file
 Each epw file has 8 lines of headers. The code reads 9 lines.
2. Use a wrong starting day
 The Tpme is supposed to be an average value in the previous 30 days. When Jan. 1 is used in the first day for simulation, the starting day for average is Dec. 2. The code uses Dec. 1.
3. Wrong calculation of Tpma
 Here is a line of calculation:
 runningAverageASH = ( 29.0 * runningAverageASH + avgDryBulbASH ) / 30.0;
 When the initial value of runningAverageASH is zero, the above calculation will give wrong output. The correct calculation is
 runningAverageASH = runningAverageASH + avgDryBulbASH / 30.0;

I test the test file using San Francisco weather file, the first day average OA temperature is the same as the calculated value from weather file.

Phase 2 fixed calculation in the rest of time.

It should be pointed out that when reading a weather file, hourly weather data are used to calculate daily average temperature. When starting a simulation, outdoor dry bulb temperature is obtained in the following lines:

		if ( BeginHourFlag ) {
			avgDryBulbASH += ( OutDryBulbTemp / 24.0 );
		}
 When BeginHourFlag is true, the current time is current hour + a time step, so that OutDryBulbTemp is not exactly same as one in the hourly weather file due to interpolation. Therefore, there are small differences in the first 30 days from program and weather file direct calculation.

Enclosed is a figure to show Running-Average-Outdoor-Air-Temperature in both years from a defect file runs. 

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [x] Code style (parentheses padding, variable names)
 - [x] Functional code review (it has to work!)
 - [x] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [x] Unit Test(s)
 - n/a IDD changes:

### CODE REVIEW

File: src/EnergyPlus/ThermalComfort.cc [hunk 1]
```diff
@@ -2515,13 +2517,13 @@ namespace ThermalComfort {
 							dryBulb = StrToReal( epwLine.substr( 0, pos ) );
 							avgDryBulbASH += ( dryBulb / 24.0 );
 						}
-						runningAverageASH = ( 29.0 * runningAverageASH + avgDryBulbASH ) / 30.0;
+						DailyAveOutTemp( i ) = avgDryBulbASH;
 					}
 				} else { // Do special things for wrapping the epw
 					calcEndDay = jStartDay;
 					calcStartDay += 365;
```
Comment (mjwitte) [inline src/EnergyPlus/ThermalComfort.cc]: Does this need to know if the epw file has 366 days for a leap year?

**Review by mjwitte — CHANGES_REQUESTED** [review]
@lgu1234 I ran the defect file with SF weather with develop ( 1c5ba897d1 ) and with this branch ( e53dcc785a ). I got the same results with both for the reported variables, and "Zone Thermal Comfort ASHRAE 55 Adaptive Model Temperature" is still -1C for all of January, in both years.

**Comment by lgu1234** [comment]
@mjwitte It is true that "Zone Thermal Comfort ASHRAE 55 Adaptive Model Temperature" is still -1C for all of January, in both years". Here is code (line 2616):

			if ( runningAverageASH >= 10.0 && runningAverageASH <= 33.5 ) {
If the above conditions are true, the temperature is calculated. Otherwise, the temperatures are -1C. Since most average outdoor temperatures are below 10C in Jan, this is why the most Adaptive Model Temperature is -1C.

I am going to check-in to allow leap year.

**Comment by mjwitte** [comment]
@lgu1234 I just realized that the reason I saw no change with this fix is because my runs were pointing to an epw file that had a .stat file present, so none of this code was active.  When I run your branch now with a lone epw (no stat file), it crashes with a debug build. I'm going to see if I can locate where.

**Comment by mjwitte** [comment]
Hmm, that's odd, EnergyPlus isn't crashing, something else is failing.

**Comment by lgu1234** [comment]
@mjwitte I am running debug version of EnergyPlus with current code. It is very slow. So far, it ran to 5/21 without crash.

**Review by mjwitte — APPROVED** [review]

**Comment by mjwitte** [comment]
@lgu1234 Retested with latest code and *without* a stat file present. Now this all looks good.  Will merge once CI completes.

**Comment by lgu1234** [comment]
@mjwitte Thanks.

**Comment by mjwitte** [comment]
linux CI failed again, merging anyway.

**Pull request merged**